redhat-developer/try-in-dev-spaces-browser-extension
# Import patternfly components from component modules

Fixes https://github.com/redhat-developer/try-in-dev-spaces-browser-extension/issues/27

The first commit helps decrease the size of the extension by not loading unnecessary Patternfly components.

ie, instead of:
```
import { Button } from "@patternfly/react-core";
```

the first commit of this PR updates Patternfly imports to
```
import { Button } from "@patternfly/react-core/components/Button";
```

The second commit helps read patternfly fonts for the production build of the extension.

The third commit helps load minified CSS (options page and content script) CSS for the production build of the extension.

These commits help reduce the size of the production build of the exntesion:
Before:
<img width="323" alt="image" src="https://user-images.githubusercontent.com/83611742/224448146-b5de33b0-1359-4b7f-ae56-f25a3b73bfee.png">

After:
<img width="322" alt="image" src="https://user-images.githubusercontent.com/83611742/224800412-e2a145bb-3538-438e-b63f-d8bd1ba8adfa.png">


## To test this PR
1. Checkout the PR branch and run `yarn && yarn build:prod` to download dependencies and start the production build.
2. Verify that the size of the extension located in `dist/chromium` is about 3MB.
3. Sideload the extension in Google Chrome and verify that the options page and injected button styles are correct:

Options page:
<img width="668" alt="image" src="https://user-images.githubusercontent.com/83611742/224448298-c7e9c483-af51-4283-83d8-6d16df705de3.png">

Injected button:
<img width="425" alt="image" src="https://user-images.githubusercontent.com/83611742/224448313-45fbd856-fa91-4c55-bad7-b0ebeef8d033.png">

4. Run `yarn build` to start the development build and sideload the extension located in `dist/chromium`.
5. Verify that the styles are correct like in step 3.
